### ISSUE TITLE
Use of dbm is problematic for cross-compilation

### ISSUE THREAD

**Issue by smcv** [lathiat/avahi#289]
The service type database `service-types.db` compiled from `service-type-database/service-types` is installed as a GNU gdbm database, or some other dbm-derived "hash-on-disk" database. Unfortunately, the format of these databases is architecture-specific: it depends on at least endianness and word size, and also appears to differ between some of the architectures supported in Debian even though their endianness and word size is the same, for example amd64 and ppc64el.

This means that if you are cross-compiling - for example, if you have an amd64 build system and you want Avahi binaries for a mips host system - you have to be able to run a mips program to generate a mips-compatible `service-types.db`. In practice this is currently done by running the host system python, for example `python3:mips` in Debian notation. In a cross-compilation scenario, this is not always possible.

There are currently only 107 non-blank, non-comment lines in `service-types`, so perhaps using dbm is overkill, and it would be sufficient to install `service-types` into `${datadir}` and adapt the `avahi-python`, `avahi-ui` and `avahi-utils` modules to parse it line-by-line?

Downstream bug report: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=873596#30

**Comment by pemensik** [comment]
Yes, I agree using any external dbm database backend for just 107 string key to string value database is overkill. Especially when we can store it sorted already and use some kind of faster searching, using sorted data in advance. If we have python building script, we could prepare it by few python functions to be easier consumable. I proposed trivial ini style for this in #721. But because key is expected to be without spaces and special characters, it could be even plain text file with format:

``_example._tcp: Name of service``

The only features we want is getting ideally localised Name of service from simple key. Current code is duplicated once in python in avahi-python/avahi/ServiceTypeDatabase.py.in, second time native C code in avahi-utils/stdb.c. Both very minimalistic implementations.

If we have cache implementation, we should have some searching tree algorithm implemented already. External database is used here to allow python and native use common data. I think it should be done simpler way, yes.

**Comment by pemensik** [comment]
I agree with Helmut on https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=873596#41. We could make these services built-in into avahi-browse by generating simple C array. The only question is how to share those types between native and python. If we make it built-in in native, then python would need some way to get that data into python.

I think there should be some way to make it extensible by user. That is why I proposed INI style parsing, which are implemented in both.

**Comment by pemensik** [comment]
If we have it built-in somehow, it should be exposed by some public API too.
